walkers/src/map.rs

```diff
@@ -376,3 +398,45 @@ fn flood_fill_tiles(
         }
     }
 }
+
+/// Implementation of the scale computation.
+fn calculate_meters_per_pixel(latitude: f64, zoom: f64, tile_size: u32) -> f64 {
+    const EARTH_CIRCUMFERENCE: f64 = 40_075_016.686;
+
+    // Number of pixels for width of world at this zoom level
+    let num_tiles = 2f64.powf(zoom);
+    let pixels_per_tile = tile_size as f64;
+    let total_pixels = num_tiles * pixels_per_tile;
```

Comment:
This one is tricky. Different tile sources can have different tile sizes (and consequently what zoom means), but we also have our own. We need to do some calculation because of that, but this way we can work without any tile source and zoom that works the same between sources.

![image](https://github.com/user-attachments/assets/6cc298d3-6122-4cd6-8fd2-4f9a1325b82d)
![image](https://github.com/user-attachments/assets/bbe63f18-af94-4cdb-bdfa-89e20d178ef5)

I think it work just fine by simply using [total_pixels](https://github.com/podusowski/walkers/blob/main/walkers/src/mercator.rs#L62)

Reply:
Well spotted! I've simplified the code by using the existing `total_pixels()` and indeed the calculation is now consistent across map types.

<img width="312" alt="image" src="https://github.com/user-attachments/assets/1d1ebbd0-0cd3-47e2-801e-7e053a11eafb">
<img width="278" alt="image" src="https://github.com/user-attachments/assets/8cf9b824-d07f-4345-9623-7cbec0e999a2">